contracts/messaging/framework/SynMessagingReceiver.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import "../interfaces/ISynMessagingReceiver.sol";
+import "../interfaces/IMessageBus.sol";
+import "@openzeppelin/contracts-4.5.0/access/Ownable.sol";
+
+abstract contract SynMessagingReceiver is ISynMessagingReceiver, Ownable {
+    
+    address public messageBus;
+
+    // Maps chain ID to the bytes32 trusted addresses allowed to be source senders
+    mapping(uint256 => bytes32) internal trustedRemoteLookup;
+
+    event SetTrustedRemote(uint256 _srcChainId, bytes32 _srcAddress);
+
+
+    /**
+     * @notice Executes a message called by MessageBus (MessageBusReceiver)
+     * @dev Must be called by MessageBug & sent from src chain by a trusted srcApp
+     * @param _srcAddress The bytes32 address of the source app contract
+     * @param _srcChainId The source chain ID where the transfer is originated from
+     * @param _message Arbitrary message bytes originated from and encoded by the source app contract
+     * @param _executor Address who called the MessageBus execution function
+     * @return status Enum containing options of Success, Fail, Retry
+     */
+    function executeMessage(
+        bytes32 _srcAddress,
+        uint256 _srcChainId,
+        bytes calldata _message,
+        address _executor
+    ) external override returns (MsgExecutionStatus) {
+        // Must be called by the MessageBus/MessageBus for security
+        require(msg.sender == messageBus, "caller is not message bus");
+        // Must also be from a trusted source app
+        require(_srcAddress == trustedRemoteLookup[_srcChainId], "Invalid source sending app");
```

Comment:
Do we ever want to have more than one trusted address on source chain?

Reply:
I think as a framework it makes sense to as an example reduce complexity & enforce only one app per source chain, but this can obviously be modified by any other apps building around it. Key point here is to clearly emphasize what typical permissioning looks like.